x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/index.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  ChoroplethMap,
+  getProgressionColor,
+  getDataRange
+} from './ChoroplethMap';
+import { ColorProgressionBar } from './ColorProgressionBar';
+import { useAvgDurationByCountry } from '../../../../../hooks/useAvgDurationByCountry';
+
+const RegionMapChartToolTip: React.SFC<{
+  geojsonProperties: { [name: string]: any };
+  data?: { key: string; value: number; doc_count?: number };
+}> = ({ geojsonProperties, data }) => {
+  return (
+    <div style={{ textAlign: 'center' }}>
+      <div style={{ fontSize: 'larger' }}>{geojsonProperties.name}</div>
+      <div>
+        {i18n.translate(
+          'xpack.apm.metrics.pageLoadCharts.RegionMapChart.ToolTip.avgPageLoadDuration',
+          {
+            defaultMessage: 'Avg. page load duration:'
+          }
+        )}
+      </div>
+      <div style={{ fontWeight: 'bold', fontSize: 'larger' }}>
+        {data && data.value}ms
+      </div>
+      <div>
+        (
+        {i18n.translate(
+          'xpack.apm.metrics.pageLoadCharts.RegionMapChart.ToolTip.countPageLoads',
+          {
+            values: { docCount: data && data.doc_count },
+            defaultMessage: '{docCount} page loads'
+          }
+        )}
+        )
+      </div>
+    </div>
+  );
+};
+
+export const RegionMapChart: React.SFC = () => {
+  const { data } = useAvgDurationByCountry();
+  const [min, max] = getDataRange(data);
+
+  return (
+    <div>
+      <ChoroplethMap
+        initialMapboxOptions={{
+          zoom: 0.85,
+          center: { lng: 0, lat: 30 }
+        }}
+        initialGeojsonSource="https://vector.maps.elastic.co/files/world_countries_v1.geo.json?elastic_tile_service_tos=agree&my_app_name=ems-landing&my_app_version=7.2.0"
+        geojsonKeyProperty="iso2"
+        data={data}
+        renderTooltip={RegionMapChartToolTip}
```

Comment:
Would it be possible to make all props except `data` constants inside `ChoroplethMap`?

```ts
<ChoroplethMap data={data}>
```

Reply:
I think we don't really know what "re-usable" means yet. Take a look at the embeddable Maps component. It's reusable but not to our use-case unfortunately. But it does have a lot of things we don't need ;)

So I think it's better to build something with the goal of making it as simple as possible, and then later _if_ we need it, it'll be easy to extend it and make it reusable.

I think `data` makes sense as a prop since it's not static. We still need to react to changes to `data` inside `useEffect` hooks and so forth. We don't need to do that for the others if they are constants inside the component.